File: data_structures/linked_list/go/linked_list.go
package main

import "fmt"

type Node struct {
	Value string
	Next  *Node
}

type LinkedList struct {
	Head   *Node
	Length int
}

func (ll *LinkedList) Insert(n *Node) bool {
	if ll.Head == nil {
		ll.Head = n
		ll.Length++
		return true
	}

	aux := ll.Head
	for aux.Next != nil {
		aux = aux.Next
	}

	aux.Next = n
	ll.Length++
	return true
}

func (ll *LinkedList) Display() {
	aux := ll.Head

	for aux != nil {
		fmt.Println(aux.Value)
		aux = aux.Next
	}
}

func main() {
	l := &LinkedList{}

	words := []string{
		"hola",
		"mundo",
		"laboratoria",
	}

	for _, value := range words {
		n1 := &Node{Value: value}
		l.Insert(n1)
	}

	l.Display()
}

File: data_structures/linked_list/python/linked_list.py
class Node():
    def __init__(self, value):
        self.value = value
        self.next = None

class LinkedList():
    def __init__(self):
        self.head = None
        self.length = 0

    def insert(self, value):
        if self.head == None:
            self.head = Node(value)
            self.length += 1
            return

        aux = self.head
        while (aux.next != None):
            aux = aux.next

        aux.next = Node(value)
        self.length += 1
        return

    def display(self):
        aux = self.head
        while (aux):
            print aux.value
            aux = aux.next

# main program
ll = LinkedList()

ll.insert(1)
ll.insert(3)
ll.insert(4)
ll.insert(2)

ll.display()
